packages/blockchain-wallet-v4/src/redux/payment/eth/sagas.js

```diff
@@ -120,12 +125,29 @@ export default ({ api }) => {
         return makePayment(merge(p, { from, effectiveBalance }))
       },
 
+      * fee (value) { // value is in gwei
+        const feeInGwei = value
+        const gasLimit = path(['fees', 'gasLimit'], p)
+        const fee = calculateFee(feeInGwei, gasLimit)
+        const accountR = yield select(S.kvStore.ethereum.getDefaultAddress)
+        const account = accountR.getOrFail('missing_default_from')
+        yield put(A.data.ethereum.fetchLegacyBalanceLoading())
+        const data = yield call(api.getEthereumBalances, account)
+        const balance = path([account, 'balance'], data)
+        yield put(A.data.ethereum.fetchLegacyBalanceSuccess(balance))
+        let effectiveBalance = calculateEffectiveBalance( // balance + fee need to be in wei
```

Comment:
Can you move this to a calculateEffectiveBalance call like btc?

Reply:
If you mean can we do `yield call` I don't believe we can.`calculateEffectiveBalance` for eth is pretty different (and simpler) than the btc version, and doesn't return a promise. But also, why would we want that?